.github/workflows/treadmill-ci.yml

```diff
@@ -139,6 +154,25 @@ jobs:
       libtock-c-ref: ${{ github.event_name == 'workflow_dispatch' && inputs.libtock-c-ref || 'master' }}
 
       # Pass the selected tests:
-      tests-json: ${{ (github.event_name == 'workflow_dispatch' && inputs.tests-json != '') && inputs.tests-json || needs.determine-tests.outputs.hwci-tests-json }}
+      tests-json: ${{ (github.event_name == 'workflow_dispatch' && inputs.tests-json != '') && inputs.tests-json || needs.hwci-determine-tests.outputs.hwci-tests-json }}
 
     secrets: inherit
+
+  # We cannot depend on *all* test-execute jobs of hwci-treadmill-dispatch as
+  # required checks for pull requests and merge queues. Thus, we run another
+  # single dummy step here that waits for all the hwci-treadmill-dispatch jobs
+  # to complete and report success.
+  #
+  # We also use this to report a "dummy" success value for the "pull_request"
+  # trigger, as explained in the comment of the "on:" parameters above.
+  hwci-report-success:
+    needs: [hwci-determine-tests, hwci-treadmill-dispatch]
+
+    if: always()
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Fail if any of the 'hwci-treadmill-dispatch' jobs failed
+        if: github.event_name != 'pull_request' && contains(needs.*.result, 'failure')
```

Comment:
On the github issue you linked on slack, I saw this example someone used:
```
if: ${{ github.event_name == 'merge_group' }}
    steps:
      - uses: actions/checkout@v2
      - name: Skip UI Tests
        uses: Sibz/github-status-action@v1
        with:
          authToken: ${{ secrets.GITHUB_TOKEN }}
          context: 'UI Tests'
          description: 'Skipped for merge queue'
```

I like that this updates the description to indicate that the tests were not actully run, which might prevent confusion for folks who see the check pass on their PR then fail in the merge queue. What do you think?

Reply:
Oh wait, now I think I understand. This is not the check that shows up in the list on a PR anyway, since it is not a required check itself. and hwci-determine-tests will show as skipped on PRs?

Reply:
I didn't actually think about this, but indeed, both `hwci-determine-tests` and `hwci-treadmill-dispatch` show as skipped. So I think the output is quite reasonable as is. :)

Reply:
And indeed, the required check is the synthetic `report-success` job, because the `hwci-treadmill-dispatch` job can actually expand into multiple jobs with different, unpredictable names, so we can't depend on them. We have to collect all of them into a single job, and mark that one as required.

Reply:
Works for me